include/fmt/core.h

```diff
@@ -740,19 +740,32 @@ inline auto get_container(std::back_insert_iterator<Container> it)
 }
 
 template <typename Char, typename InputIt, typename OutputIt>
-FMT_CONSTEXPR auto copy_str(InputIt begin, InputIt end, OutputIt out)
+FMT_CONSTEXPR auto copy_str_constexpr(InputIt begin, InputIt end, OutputIt out)
     -> OutputIt {
   while (begin != end) *out++ = static_cast<Char>(*begin++);
   return out;
 }
 
-template <typename Char, FMT_ENABLE_IF(std::is_same<Char, char>::value)>
-FMT_CONSTEXPR auto copy_str(const Char* begin, const Char* end, Char* out)
-    -> Char* {
+template <typename Char, typename InputIt, typename OutputIt>
+FMT_CONSTEXPR auto copy_str(InputIt begin, InputIt end, OutputIt out)
+    -> OutputIt {
+  return copy_str_constexpr<Char>(begin, end, out);
+}
+
+template <typename Char, typename InputIt, typename OutputIt>
+FMT_CONSTEXPR
+typename std::enable_if
+<
+  std::is_same<typename std::remove_const<InputIt>::type, OutputIt>::value &&
+  std::is_trivially_copy_assignable<OutputIt>::value,
+  OutputIt*
+>::type
+copy_str(InputIt* begin, InputIt* end, OutputIt* out)
+{
   if (is_constant_evaluated())
-    return copy_str<Char, const Char*, Char*>(begin, end, out);
```

Comment:
This will mean that we cannot call `copy_str` here but it's just a one-line, so can be just copy the while loop here (and remove the cast).

Reply:
I don't see a problem here.